### PR TITLE
Refactor fid reader

### PR DESCRIPTION
**Pull request by lingwei-gu**
Notebook:
https://colab.research.google.com/drive/1DLX59bJvFjdv3wQM6wU0DDlcsaJzIAJ-#scrollTo=69zZa1KEPp5U

### CODE REVIEW

**Review by ronakice — CHANGES_REQUESTED** [review]
Add these for now I'll add more comments in a while

File: pygaggle/qa/fid/model.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
```
[the remark this answers is not in the document]
Comment (lingwei-gu) [reply pygaggle/qa/fid/model.py]: added the link to the license

File: pygaggle/qa/fid_reader.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
+import sys
+sys.path.insert(1, '/u0/o3liu/FiD')
```
Comment (ronakice) [inline pygaggle/qa/fid_reader.py:2]: what's this?

File: pygaggle/qa/fid_reader.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        self.batch_size = batch_size
+
+    # expanded upon Collator used by DataLoader
+    def build_dataloader(
+            self,
+            question: Question,
+            contexts: List[Context], #corresponde to passages in fid
+    ):
+        # def encode_passages(batch_text_passages, tokenizer, max_length):
+        #     passage_ids, passage_masks = [], []
```
Comment (ronakice) [inline pygaggle/qa/fid_reader.py:50]: can we remove unnecessary comments?

File: pygaggle/qa/fid_reader.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+            #         input_ids=context_ids.cuda(),
+            #         attention_mask=context_mask.cuda(),
+            #         max_length=50
+            #     )
+            #     for k, o in enumerate(outputs):
+            #         ans = self.tokenizer.decode(o, skip_special_tokens=True)
+
+        
+if __name__ == "__main__":
+    print("main function executed")
```
Comment (ronakice) [inline pygaggle/qa/fid_reader.py:148]: just this in main?

File: pygaggle/run/evaluate_fid_ranker.py [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+from pygaggle.qa.span_selection import DprSelection, GarSelection, DprFusionSelection, GarFusionSelection
+from pygaggle.data.retrieval import RetrievalExample
+from typing import List, Optional, Dict
+from tqdm import tqdm
+import numpy as np
+import string
+import regex as re
+
+
+READER_CHOICES = ('dpr', 'fid')
```
Comment (ronakice) [inline pygaggle/run/evaluate_fid_ranker.py:23]: we don't need this choice here, since only fid exists.

**Comment by ronakice** [comment]
Okay, thanks! Merging!

**Review by ronakice — APPROVED** [review]
LGTM!

**Pull request merged**